src/components/pagination/index.tsx

```diff
@@ -0,0 +1,270 @@
+								// This key will be stable between re-renders
+								// which prevents react from removing focus from
+								// the previous `document.activeElement`
+								key={e}
+							>
+								<ButtonNumber page={e}>{e}</ButtonNumber>
+							</li>
+						)
+					})}
+				</ul>
+			)}
+			<ButtonArrow type={type} direction="next" />
+		</nav>
+	)
+}
+
+const ButtonArrow = ({ type, direction }: ButtonArrowProps) => {
+	const pagination = usePagination()
+	const handleClick = {
+		next: () => {
+			// clamp upper bound
+			const nextIdx = Math.min(pagination.page + 1, pagination.totalPages)
+			pagination.setPage(nextIdx)
+			pagination.onPageChange(nextIdx)
+		},
+		prev: () => {
+			// clamp lower bound (0)
+			const prevIdx = Math.max(pagination.page - 1, 0)
+			pagination.setPage(prevIdx)
+			pagination.onPageChange(prevIdx)
+		},
+	}[direction]
+
+	const icon = {
+		next: <IconChevronRight16 />,
+		prev: <IconChevronLeft16 />,
+	}[direction]
+
+	const ariaLabel = {
+		next: 'Next page',
+		prev: 'Previous page',
+	}[direction]
+
+	const label = {
+		next: 'Next',
+		prev: 'Previous',
+	}[direction]
+
+	const isDisabled = {
+		next: pagination.page === pagination.totalPages,
+		prev: pagination.page === 1,
+	}[direction]
+
+	return (
+		<button
+			className={classNames(
+				s.arrow,
+				s.control,
+				{
+					next: s.next,
+					prev: s.prev,
+				}[direction]
+			)}
+			aria-label={ariaLabel}
+			onClick={handleClick}
+			disabled={isDisabled}
+		>
+			{icon}
+			{type === 'compact' && (
+				<span className={classNames(s.label)} aria-hidden="true">
+					{label}
+				</span>
+			)}
+		</button>
+	)
+}
+
+const ButtonNumber = ({
+	children,
+	page,
+}: PropsWithChildren<ButtonNumberProps>) => {
+	const pagination = usePagination()
+	const handleClick = () => {
+		pagination.setPage(page)
+		// don't trigger callback if the current page is already selected
+		if (pagination.page !== page) {
+			pagination.onPageChange(page)
+		}
+	}
+	return (
+		<button
+			className={classNames(s.control)}
+			type="button"
+			onClick={handleClick}
+		>
+			<span className={'g-screen-reader-only'}>page&nbsp;</span>
+			{children}
+		</button>
+	)
+}
+
+const SizeSelector = ({ sizes }: SizeSelectorProps) => {
+	const pagination = usePagination()
+
+	// changing page size should reset the current page to 1
+	const handleChange = (e) => {
+		const newPageSize = Number(e.target.value)
+		// don't trigger callbacks if pagesize didn't change
+		if (pagination.pageSize !== newPageSize) {
+			pagination.setPage(1)
+			pagination.setPageSize(newPageSize)
+			pagination.onPageSizeChange(newPageSize)
+			pagination.onPageChange(1)
+		}
+	}
+
+	return (
+		<div className={s['size-selector']}>
+			<label className={s.label}>
+				Items per page
+				<select className={s.select} onChange={handleChange}>
+					{sizes.map((e) => (
+						<option key={e} value={e} selected={pagination.pageSize === e}>
+							{e}
+						</option>
+					))}
+				</select>
```

Comment:
🙋🏻 Can the `e` variable name be updated to `size` for clarity?

```suggestion
				<select className={s.select} onChange={handleChange}>
					{sizes.map((size) => (
						<option
							key={size}
							value={size}
							selected={pagination.pageSize === size}
						>
							{size}
						</option>
					))}
				</select>
```

Reply:
Updated!